client/src/main/java/com/aliyun/emr/rss/client/compress/RssZstdCompressor.java

```diff
@@ -69,6 +69,8 @@ public void compress(byte[] data, int offset, int length) {
       compressMethod = COMPRESSION_METHOD_ZSTD;
     }
 
+    int level = compressionLevel < 0 ? (Math.abs(compressionLevel) + 22) : compressionLevel;
+    assert (0 <= level && level <= 27);
     compressedBuffer[MAGIC_LENGTH] = (byte) (compressMethod | compressionLevel);
```

Comment:
I want to know why need this logic? seems only 
```
compressedBuffer[MAGIC_LENGTH] = compressMethod 
```
can work? 

@waitinfuture @FMX

Reply:
The decompressor needn't know `compressionLevel` IMO

Reply:
Agree with @AngersZhuuuu and @pan3793 . After checking history I found the following:
1. the reason lz4 encodes compressionLevel is to follow logic in net.jpountz.lz4.LZ4BlockOutputStream.flushBufferedData:
```
    compressedBuffer[MAGIC_LENGTH] = (byte) (compressMethod | compressionLevel);
```
However, the compressionLevel are used in LZ4BlockInputStream.refill to do validation:
```
    final int compressedLen = SafeUtils.readIntLE(compressedBuffer, MAGIC_LENGTH + 1);
    originalLen = SafeUtils.readIntLE(compressedBuffer, MAGIC_LENGTH + 5);
    final int check = SafeUtils.readIntLE(compressedBuffer, MAGIC_LENGTH + 9);
    assert HEADER_LENGTH == MAGIC_LENGTH + 13;
    if (originalLen > 1 << compressionLevel
        || originalLen < 0
        || compressedLen < 0
        || (originalLen == 0 && compressedLen != 0)
        || (originalLen != 0 && compressedLen == 0)
        || (compressionMethod == COMPRESSION_METHOD_RAW && originalLen != compressedLen)) {
```
but RssLz4Decompressor has remove the check logic, so I think either we can add back the check, or we can delete encode the level in compressedBuffer.
2. Although in RssLz4Decompressor the compression level is redundant, it is safe since Lz4 ensures the compression level will not be larger than 0xF:
```
    compressionLevel = Math.max(0, compressionLevel - COMPRESSION_LEVEL_BASE);
    assert compressionLevel >= 0 && compressionLevel <= 0x0F;
```
```
   * @param blockSize   the maximum number of bytes to try to compress at once,
   *                    must be &gt;= 64 and &lt;= 32 M
```
3. But in RssZstdCompressor it's dangerous to encode compressed level, because it can exceeds 0xF.

So IMO at least we should delete compressed level in RssZstdCompressor. @AngersZhuuuu @pan3793 @nafiyAix

Reply:
Thanks, get it. And in RssLz4Compressor seems it doesn't use the compression level when compressing, could we remove it completely from this class?

Reply:
> Thanks, get it. And in RssLz4Compressor seems it doesn't use the compression level when compressing, could we remove it completely from this class?

I think so.